OpenQA/Isotovideo/Utils.pm

```diff
@@ -71,4 +74,28 @@ sub checkout_git_repo_and_branch {
     return $bmwqemu::vars{$dir_variable} = File::Spec->rel2abs($local_path);
 }
 
+=head2 checkout_git_refspec
+
+    checkout_git_refspec($dir, $refspec_variable);
+
+Takes a git working copy directory path and checks out a git refspec specified
+in a git hash test parameter if possible. Returns the determined git hash in
+any case, also if C<$refspec> was not specified or is not defined.
+
+Example:
+
+    checkout_git_refspec('/path/to/casedir', 'TEST_GIT_REFSPEC');
+
+=cut
+sub checkout_git_refspec {
+    my ($dir, $refspec_variable) = @_;
+    return undef unless defined $dir;
+    if (my $refspec = $bmwqemu::vars{$refspec_variable}) {
+
+        diag "Checking out local git refspec '$refspec' in '$dir'";
+        qx{env git -C $dir checkout -q $refspec};
```

Comment:
Have you tested what happens when this Git command fails?

Reply:
Yes, for sure. I thought about having it ignore if the checkout fails until we have gathered some experience but that's not a good idea. I added a `die` now after the call if the command fails.